samples/v2/precompiled/RestartVMs.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace VSSample
+{
+    // To authorize ARM calls, your subscription has to have permissions to your function app.
+    // Making the subscription an owner of your function app is one solution to this.
+    public static class RestartVMs
+    {
+
+        [FunctionName("RestartVMs")]
+        public static async Task RunOrchestrator(
+            [OrchestrationTrigger] IDurableOrchestrationContext context,
+            ILogger log)
+        {
+            string apiVersion = await context.CallActivityAsync<string>("GetApiVersion", null);
+            string subscriptionId = await context.CallActivityAsync<string>("GetSubscriptionId", null);
+            string resourceGroup = await context.CallActivityAsync<string>("GetResourceGroup", null);
+
+            ManagedIdentityTokenSource managedIdentityTokenSource = new ManagedIdentityTokenSource("https://management.core.windows.net");
+
+            // List all of the VMs in my subscription and add them to an ArrayList
+            string listAllCallString = $"https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/virtualMachines?api-version={apiVersion}";
+            Uri listAllUri = new Uri(listAllCallString);
+            DurableHttpRequest listRequest = new DurableHttpRequest(method: HttpMethod.Get, uri: listAllUri, tokenSource: managedIdentityTokenSource);
+            DurableHttpResponse listAllResponse = await context.CallHttpAsync(listRequest);
+
+            // Deserializes content to just get the names of the VMs in the subscription
+            JObject jObject = JsonConvert.DeserializeObject<JObject>(listAllResponse.Content);
+            ArrayList vmNamesList = new ArrayList();
+            foreach (var value in jObject["value"])
+            {
+                string vmName = value["name"].ToString();
+                vmNamesList.Add(vmName);
+            }
+
+            // Restart all of the VMs in my subscription
+            foreach (string vmName in vmNamesList)
+            {
+                var restartVMCallString = $"https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachines/{vmName}/restart?api-version={apiVersion}";
+                DurableHttpRequest restartRequest = new DurableHttpRequest(method: HttpMethod.Post, uri: new Uri(restartVMCallString), tokenSource: managedIdentityTokenSource);
+                DurableHttpResponse restartResponse = await context.CallHttpAsync(restartRequest);
+            }
+        }
+
+        [FunctionName("GetApiVersion")]
+        public static string GetApiVersion([ActivityTrigger] string name, ILogger log)
+        {
+            // Get API Version from environment variables
+            return Environment.GetEnvironmentVariable("ApiVersion", EnvironmentVariableTarget.Process);
```

Comment:
It's better to use the App Settings APIs. That way things work with both app settings, local.settings.json, and environment variables